thumbor/engines/pil.py

```diff
@@ -275,6 +282,14 @@ def read(self, extension=None, quality=None):  # NOQA
 
         results = img_buffer.getvalue()
         img_buffer.close()
+
+        if METADATA_AVAILABLE and self.context.config.PRESERVE_IMAGE_METADATA:
+            results_metadata = ImageMetadata.from_buffer(results)
+            results_metadata.read()
+            self.metadata.copy(results_metadata)
```

Comment:
I am afraid it's a bit more complex than that. 
If image was rotated according to exif, we should change that in exif written here, otherwise image will have incorrect orientation.
If request called for strip_icc, this code should also remove icc profile.

Reply:
I can disable the copy of exif (by passing exif=False in ImageMetadata copy method). So the exif will be the same that was setting in the image object.

In my tests, the icc_profile will be keep from image object because de ImageMetadata does not handle icc profile.